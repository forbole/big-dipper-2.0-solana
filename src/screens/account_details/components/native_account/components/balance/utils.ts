import { formatNumber } from '@utils/format_token';
import { BalanceType } from './types';

export const formatBalanceData = (data: BalanceType) => {
  const balanceChart = [
    {
      key: 'balanceNative',
      display: `${formatNumber(data.native.value, data.native.exponent)} ${data.native.displayDenom.toUpperCase()}`,
      value: data.native.value,
    },
    {
      key: 'balanceStake',
      display: `${formatNumber(data.stake.value, data.stake.exponent)} ${data.stake.displayDenom.toUpperCase()}`,
      value: data.stake.value,
    },
    {
      key: 'balanceNonce',
      display: `${formatNumber(data.nonce.value, data.nonce.exponent)} ${data.nonce.displayDenom.toUpperCase()}`,
      value: data.nonce.value,
    },
    {
      key: 'balanceVote',
      display: `${formatNumber(data.vote.value, data.vote.exponent)} ${data.vote.displayDenom.toUpperCase()}`,
      value: data.vote.value,
    },
  ];

  return balanceChart;
};
