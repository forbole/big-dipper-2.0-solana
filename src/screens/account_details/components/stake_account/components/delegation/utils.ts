import numeral from 'numeral';
import { DelegationType } from './types';

export const formatBalanceData = (data: DelegationType) => {
  const balanceChart = [
    {
      key: 'balanceActivated',
      display: `${numeral(data.activated.value).format(data.activated.format)} ${data.activated.denom.toUpperCase()}`,
      value: data.activated.value,
    },
    {
      key: 'balanceActivating',
      display: `${numeral(data.activating.value).format(data.activating.format)} ${data.activating.denom.toUpperCase()}`,
      value: data.activating.value,
    },
  ];

  return balanceChart;
};
