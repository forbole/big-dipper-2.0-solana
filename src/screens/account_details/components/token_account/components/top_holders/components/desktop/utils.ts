export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 5,
    align: 'center',
  },
  {
    key: 'holder',
    width: 65,
  },
  {
    key: 'amount',
    width: 30,
    align: 'right',
  },
];
