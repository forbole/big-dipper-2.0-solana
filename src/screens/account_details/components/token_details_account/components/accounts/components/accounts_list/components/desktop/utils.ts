export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'address',
    width: 95,
  },
];
