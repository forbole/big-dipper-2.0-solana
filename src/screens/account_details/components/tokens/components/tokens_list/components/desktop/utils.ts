export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'token',
    width: 50,
  },
  {
    key: 'amount',
    align: 'right',
    width: 50,
  },
];
