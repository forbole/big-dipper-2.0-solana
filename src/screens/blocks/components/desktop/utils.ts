export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'slot',
    width: 10,
  },
  {
    key: 'leader',
    width: 35,
  },
  {
    key: 'hash',
    width: 30,
  },
  {
    key: 'txs',
    align: 'right',
    width: 10,
  },
  {
    key: 'time',
    align: 'right',
    width: 15,
  },
];
