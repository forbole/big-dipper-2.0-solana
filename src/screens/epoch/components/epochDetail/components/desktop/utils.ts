export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'name',
    width: 30,
  },
  {
    key: 'value',
    width: 20,
  },
  {
    key: 'detail',
    width: 50,
  },
];
