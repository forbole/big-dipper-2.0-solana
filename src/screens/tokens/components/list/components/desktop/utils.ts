export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'idx',
    width: 15,
  },
  {
    key: 'token',
    width: 35,
  },
  {
    key: 'price',
    width: 25,
    align: 'right',
  },
  {
    key: 'marketCap',
    align: 'right',
    width: 25,
  },
];
