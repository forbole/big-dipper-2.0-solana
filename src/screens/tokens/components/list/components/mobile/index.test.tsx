import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Mobile from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  SingleBlockMobile: (props) => <div id="SingleBlockMobile" {...props} />,
  Loading: (props) => <div id="Loading" {...props} />,
  AvatarName: (props) => <div id="AvatarName" {...props} />,
}));

jest.mock('react-virtualized-auto-sizer', () => ({ children }: any) => children({
  height: 600, width: 600,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Home/Blocks/Mobile', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Mobile
          items={[
            {
              token: 'BTC',
              price: 1802,
              marketCap: 515912496,
              volume: 18333060,
              address: '9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E',
              logo: 'https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/G2jrxYSoCSzmohxERa2JzSJMuRM4kiNvRA3DnCv7Lzcz/logo.png',
            },
          ]}
          itemCount={2}
          loadMoreItems={() => jest.fn()}
          isItemLoaded={() => true}
        />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
