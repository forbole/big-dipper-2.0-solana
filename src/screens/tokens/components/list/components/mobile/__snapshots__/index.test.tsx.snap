// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Blocks/Mobile matches snapshot 1`] = `
<div
  className="makeStyles-root"
>
  <div
    className="List"
    onScroll={[Function]}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 600,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        Object {
          "height": 200,
          "pointerEvents": undefined,
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "height": 100,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": "100%",
          }
        }
      >
        <div>
          <div
            className="makeStyles-root"
          >
            <div
              className="makeStyles-item"
            >
              <h4
                className="MuiTypography-root label MuiTypography-h4"
              >
                tokens:token
              </h4>
              <a
                className="MuiTypography-root value MuiTypography-body1"
                href="/tokens/9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                BTC
              </a>
            </div>
            <div
              className="makeStyles-item"
            >
              <h4
                className="MuiTypography-root label MuiTypography-h4"
              >
                tokens:price
              </h4>
              <p
                className="MuiTypography-root value MuiTypography-body1"
              >
                $1,802
              </p>
            </div>
            <div
              className="makeStyles-item"
            >
              <h4
                className="MuiTypography-root label MuiTypography-h4"
              >
                tokens:marketCap
              </h4>
              <p
                className="MuiTypography-root value MuiTypography-body1"
              >
                $515,912,496
              </p>
            </div>
          </div>
          <hr
            className="MuiDivider-root"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "height": 100,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 100,
            "width": "100%",
          }
        }
      >
        <div>
          <div
            className="makeStyles-root"
          >
            <div
              className="makeStyles-item"
            >
              <h4
                className="MuiTypography-root label MuiTypography-h4"
              >
                tokens:token
              </h4>
            </div>
            <div
              className="makeStyles-item"
            >
              <h4
                className="MuiTypography-root label MuiTypography-h4"
              >
                tokens:price
              </h4>
              <p
                className="MuiTypography-root value MuiTypography-body1"
              />
            </div>
            <div
              className="makeStyles-item"
            >
              <h4
                className="MuiTypography-root label MuiTypography-h4"
              >
                tokens:marketCap
              </h4>
              <p
                className="MuiTypography-root value MuiTypography-body1"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
